src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -1495,14 +1506,44 @@ private async Task WriteRevertedRevisions(List<Document> list, OperationCancelTo
             list.Clear();
         }
 
-        private bool PrepareRevertedRevisions(DocumentsOperationContext writeCtx, Parameters parameters, RevertResult result, List<Document> list, OperationCancelToken token)
+        private bool PrepareRevertedRevisions(DocumentsOperationContext writeCtx, Parameters parameters, RevertResult result, List<Document> list, OperationCancelToken token, HashSet<string> collections = null)
         {
             using (_database.DocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext readCtx))
             using (readCtx.OpenReadTransaction())
             {
-                var revisions = new Table(RevisionsSchema, readCtx.Transaction.InnerTransaction);
-                foreach (var tvr in revisions.SeekBackwardFrom(RevisionsSchema.FixedSizeIndexes[AllRevisionsEtagsSlice],
-                    parameters.LastScannedEtag))
+                IEnumerable<Table.TableValueHolder> tvrs = null;
+                if (collections != null)
+                {
+                    foreach(var collection in collections)
+                    {
+                        if (collection == null)
+                        {
+                            if(_logger.IsInfoEnabled)
+                                _logger.Info($"Tried to revert revisions in collection that is null");
+                            continue;
+                        }
+                        var collectionName = _documentsStorage.GetCollection(collection, throwIfDoesNotExist: false);
+                        if (collectionName == null)
+                        {
+                            if (_logger.IsInfoEnabled)
+                                _logger.Info($"Tried to revert revisions in the collection '{collection}' which does not exist");
```

Comment:
We have any way to return information to the user, no? `RevertResult `

Reply:
Done